api/k8s/v1beta4/k8s_helpers.go

```diff
@@ -47,3 +47,60 @@ func (s *API) WaitForCluster(req *WaitForClusterRequest) (*Cluster, error) {
 	}
 	return cluster.(*Cluster), nil
 }
+
+func (s *API) getNodesWhenReady(req *ListNodesRequest, expectedCount uint32) (interface{}, bool, error) {
+	res, err := s.ListNodes(req, scw.WithAllPages())
+	if err != nil {
+		return nil, false, err
+	}
+
+	// first check that we have the right nodes count
+	if res.TotalCount != expectedCount {
+		return nil, false, nil
+	}
+
+	// then check that each node has Ready status
+	for _, node := range res.Nodes {
```

Comment:
and no need to check the status of the nodes, if the pool is ready, all the nodes in the pool are ready

Reply:
Are you sure of this ? I did a test using only the pool Status and it seems to be "ready" right after it is created but this does not take into account the readiness of the nodes.

Reply:
It does but when nodes exists. Right after pool creation, nodes does not exists, that why the combo target size == current node count + status works :) 
(You can try to check the status a few seconds  after the creation of the pool).

Reply:
If it doesn't work (which it should 😅) I'll fix that tomorrow :)

Reply:
OK so I would have to add the target size as a parameter to check on this combo ?

Reply:
I tried something like
```
func (s *API) WaitForPoolNodesReady(req *WaitForPoolNodesReadyRequest) error {
	poolRequest := &GetPoolRequest{
		PoolID: req.PoolID,
	}

	_, err := async.WaitSync(&async.WaitSyncConfig{
		Get: func() (interface{}, bool, error) {
			pool, err := s.GetPool(poolRequest)
			if err != nil {
				return nil, false, err
			}

			if pool.Size == pool.CurrentNodeCount && pool.Status == "ready" { // this is immediately true after creation or update
				return nil, true, nil
			} else {
				return nil, false, nil
			}
		},
		Timeout:          req.Timeout,
		IntervalStrategy: async.LinearIntervalStrategy(5 * time.Second),
	})

	return err
}
```

and it does not work. Or do I have to provide the target size as a parameter ? If yes it's not trivial when looping in WaitForClusterNodesReady so I think I will stick with "listing" the nodes to retrieve the total count and check for every status.

Reply:
Hmm this should work I guess, what behaviour did you get?

Reply:
The condition was true immediately even though I added ```pool.Size == pool.CurrentNodeCount```

Reply:
Hmm I just checked, the current node count is different than the size on creation, but status is always ready 🤔 I'll take a closer look tomorrow on this problem

Reply:
Let me test again on creation since I only did test on update (1 -> 2 nodes / 2 -> 1 node)

Reply:
But it still won't work since the status problem 😅 should be fixed tomorrow though

Reply:
Confirmed that this is the same behaviour with creation.
My Terraform output (using https://github.com/terraform-providers/terraform-provider-scaleway/pull/393) looks like this:
```
nodes = [
  {
    "public_ip" = ""
    "public_ip_v6" = ""
    "status" = "creating"
  },
]
```

Reply:
Now using the pool status